server/public/model/config.go

```diff
@@ -4672,7 +4674,13 @@ func (o *Config) GetSanitizeOptions() map[string]bool {
 	return options
 }
 
-func (o *Config) Sanitize(pluginManifests []*Manifest) {
+// Sanitize removes sensitive information from the configuration object.
+// It replaces sensitive fields with [FakeSetting] or sanitizes them.
+//
+// Parameters:
+//   - pluginManifests: Plugin manifests for sanitizing plugin settings.
+//   - partiallySanitizeDataSources: Whether to partially sanitize SQL data sources instead of fully replacing it with [FakeSetting].
+func (o *Config) Sanitize(pluginManifests []*Manifest, partiallySanitizeDataSources bool) {
```

Comment:
To avoid future breaking changes, I'd consider passing an `opts` object that we can extend if needed. It would also make it more explicit on the calling side.